robot/src/robot.py

```diff
@@ -18,9 +20,53 @@ def __init__(self):
 
         self.ultrasonic_sensor = UltrasonicSensor(Port.S4)
 
-    def obstacle_in_front(self):
+        self.routes = {1: [Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'),
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}',
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'))]}
+
+        self.event_source = event_source
+
+
+    async def set_movement_speed(self, event):
+        speed = event.data.get('argument')
+        self.drive_base.settings(speed=speed)
```

Comment:
event.data is a utf-8-encoded JSON, so to be able to treat it as a dictionary i think you need to do something like:
Top of file:
`import json`

```suggestion
    async def set_movement_speed(self, event):
        json = json.loads(event.data)
        self.drive_base.settings(speed=json['argument'])
```
But if you've tested this and it works, i guess micropython is magical.